### DISCUSSION TITLE
Array&String toolbox enhancement

### DISCUSSION THREAD

**Discussion by Rratic** [JuliaLang/julia#47975] [General]
Here's a set of things that need to be enhanced found while developing
- "find"-series functions for arrays (e.g. `findfirst(0, [0])`)
- more deleting function for vectors (delete indicated value, etc)
- string handle: splice; function to make a string end with exact amounts of `/`, etc
- enhance string replacing function

I'm using 1.8.3 and quite surprised many functions are still not in Base

ideas welcome!

**Comment by andrewjradcliffe** [comment]
- "find"-series: use predicates, e.g. curried predicates `findfirst(==(0), [0])`
- deleting functions: `filter!` with an appropriate predicate should meet the need. If value is known by index, then `deleteat!`
